R/uvhydrograph-data.R

```diff
@@ -25,6 +25,21 @@ parseCorrectionsByMonth <- function(reportObject, seriesName, month) {
   return(corrections)
 }
 
+#' Parse Secondary Corrections
+#' @description depending on the report configuration, corrections might come from a reference or upchain series
+#' @param reportObject the report to render
+#' @return corrections list from the correct series
+parseSecondaryCorrectionsByMonth <- function(reportObject, month) {
+  hasReferenceSeries <- hasReferenceSeries(reportObject)
+  hasUpchainSeries <- hasUpchainSeries(reportObject)
+  
+  if(hasReferenceSeries) {
+    corrections <- parseCorrectionsByMonth(reportObject, "referenceSeriesCorrections", month)
+  } else {
+    corrections <- parseCorrectionsByMonth(reportObject, "upchainSeriesCorrections", month)
+  }
+}
```

Comment:
Lindsay usually recommends having an explicit return at the end of functions instead of using the implicit one.